pkg/kubectl/cmd/util/helpers.go

```diff
@@ -660,3 +661,12 @@ func Warning(cmdErr io.Writer, newGeneratorName, oldGeneratorName string) {
 		oldGeneratorName,
 	)
 }
+
+// WarningWriter serves as a bridge between the standard log package and the klog package.
+type WarningWriter struct{}
+
+// Write implements the io.Writer interface.
+func (writer WarningWriter) Write(data []byte) (n int, err error) {
+	klog.WarningDepth(1, string(data))
+	return len(data), nil
+}
```

Comment:
can we use WarningWriter/KlogWriter from component-base for cmd/kubectl too?

Reply:
@neolit123 that was what i tried first, looks like stuff was copied out to separate out kubectl and importing component-base causes double-ly registered flags now. so i had to have an extra copy

Reply:
seems like a problem kubectl/component-base need to resolve.